### PR TITLE
mlog standardisation for app/channel.go

### LINKED ISSUE #12722: Refactor "app/channel.go" to use structured logging
Mattermost is refactoring the logs to be fully structured and we're looking for contributors to help with that effort. This Help Wanted issue is to change the usages of `mlog.` methods with `fmt.Sprintf` to proper `mlog.Field()` calls in file `app/channel.go`.

The expected way to implement it is to open the file `app/channel.go`, look for all calls to `mlog.Info`, `mlog.Debug`, etc and in cases where `fmt.Sprintf` is used to write the log - modify the call to use `mlog.Bool`, `mlog.String`, etc to pass the parameters, instead of passing them as part of the format string to `Sprintf`.

Example: https://github.com/mattermost/mattermost-server/pull/12038

----
If you're interested please comment here and come [join our "Contributors" community channel](https://community.mattermost.com/core/channels/tickets) on our daily build server, where you can discuss questions with community members and the Mattermost core team. For technical advice or questions, please  [join our "Developers" community channel](https://community.mattermost.com/core/channels/developers).

New contributors please see our [Developer's Guide](https://developers.mattermost.com/contribute/getting-started/).

JIRA: https://mattermost.atlassian.net/browse/MM-18282

### PR DESCRIPTION
**Pull request by bensooraj**
#### Summary
Refactoring logs to change the usages of `mlog.` methods with `fmt.Sprintf` to proper `mlog.Field()` calls in file `app/channel.go`.

#### Ticket Link
Fixes #12722

### CODE REVIEW

File: app/channel.go [hunk 1]
```diff
@@ -861,21 +861,21 @@ func (a *App) DeleteChannel(channel *model.Channel, userId string) *model.AppErr
...
-			mlog.Error(fmt.Sprintf("Failed to post archive message %v", err))
+			mlog.Error("Failed to post archive message", mlog.Err(err))
 		}
 	}
 
 	now := model.GetMillis()
 	for _, hook := range incomingHooks {
 		if err := a.Srv.Store.Webhook().DeleteIncoming(hook.Id, now); err != nil {
-			mlog.Error(fmt.Sprintf("Encountered error deleting incoming webhook, id=%v", hook.Id))
+			mlog.Error("Encountered error deleting incoming webhook", mlog.String("hook_id", hook.Id))
```
Comment (hanzei) [inline app/channel.go]: Can you please also log `err` here

File: app/channel.go [hunk 2]
```diff
@@ -861,21 +861,21 @@ func (a *App) DeleteChannel(channel *model.Channel, userId string) *model.AppErr
...
 		}
 		a.InvalidateCacheForWebhook(hook.Id)
 	}
 
 	for _, hook := range outgoingHooks {
 		if err := a.Srv.Store.Webhook().DeleteOutgoing(hook.Id, now); err != nil {
-			mlog.Error(fmt.Sprintf("Encountered error deleting outgoing webhook, id=%v", hook.Id))
+			mlog.Error("Encountered error deleting outgoing webhook", mlog.String("hook_id", hook.Id))
```
Comment (hanzei) [inline app/channel.go]: Same here

File: app/channel.go (hunk 1 above)
Comment (bensooraj) [reply app/channel.go]: Thanks for the review @hanzei!

I have made the changes and pushed my commits.

**Comment by hanzei** [comment]
@bensooraj Would you please merge `master` into your branch?

**Comment by hanzei** [comment]
Thanks for the contribution @bensooraj :tada:

**Pull request merged**